### PR TITLE
Adding TypeScript definitions for react-native-simple-animations

### PR DESCRIPTION
**Pull request by iamwinner422**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:

- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:

- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

If removing a declaration:

- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/react-native-simple-animations/react-native-simple-animations-tests.tsx
1. types/react-native-simple-animations/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against b5605bc9860a6893babaff17b204699afeba2a91
</p>

File: types/react-native-simple-animations/package.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        "react-native": "*"
+    },
+    "devDependencies": {
+        "@types/react-native-simple-animations": "workspace:."
+    },
+    "owners": [
+        {
+            "name": "Joey Schroeder",
+            "githubUsername": "joeyschroeder"
+        },
```
Comment (jakebailey) [inline types/react-native-simple-animations/package.json:19]: Please don't add people unless they explicitly ask to be added. I would doubt they want to be here.

File: types/react-native-simple-animations/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
+import { ReactNode, Component } from "react";
+import { EasingFunction, ViewStyle } from "react-native";
+
+export interface SimpleAnimationProps {
+    /**
+     * Whether the child component will animate in or out
+     * @default "in"
+     */
+    aim?: "in" | "out";
```
Comment (jakebailey) [inline types/react-native-simple-animations/index.d.ts:9]: I highly suspect that these should all have `| undefined` to support `exactOptionalPropertyTypes`.

See: [Playground Link](https://www.typescriptlang.org/play?exactOptionalPropertyTypes=true&ts=4.4.2#code/PTAEHUEsBcAtQKYA8CGBjaB5ADtSB7AOxQBsAFAJ32wQugE8AVemgZ0WICMSEATAGlDMaAZTQVIuALAAoEKFgp2raBIy1QFAK492KKlsK9QAd0XRQ9fFtBoUhUKwQX7oaniKlZ87FRp16UGh8ADpZbzAAMXwKRFQAW2weQThIdkhCaFoAM3QEUF4ENBJ9BHY4FAs4Wny00BQI0F9qWgZTWEg0eDsHTnyifuygk3xHVQyAc1YALlAAcl59AGs5t1i5kkgJ2Gg5xpjQGEdYaxJjQnwLPsOHardOACsi6DCZWQysily0fIBVJwoABEELkdNB2ABvWSgUDyRiwfIoThOQg-NxDFCgABupC0+QoCF8ZQQmXYc1Y9BUCHiexkMLQ+BIMXh1IQmCxtAkhQA-LMAESLChLPmgAA+oD5m220D5AG5ZABfcJyMBQODWKoddjZEoTDhIngCIIIgn1U1wAn5HEkPHsMydeBWGw9RpOCwAAwZTIoLPibI5FC5CHdQVGfX5guFgklWx2IvsxndhkK2QyfHdr0aImceEIeru1rxodASaMILTvBDJkgJBI9Vr+BMoHi+BUoAAUigcWIJLhNIY8H7Gl0ikttQc7sg0llUflgqBcjXBPauod2CDss9IBySIFciQKa95AAJRsIAMpE0Icn1BdaABe97TrHKo1S7D6iixBAoISECIKctCBgAhCEabIDk9RlmQRP12U5SBChDAB9Xh8DKQg5mgZCbmKLRChLZMgPTTMVX-fIdRQPV3WQdAsFwUDSEoFoAmEMoQ3iFAljKIQWAQHtJBcNA0C0ChKgQXdGk46BVzuFMMhAogmiob9CmMJR6gcdxGLrZp-AYUjCmKUo71RDwHAmZx-lobNoFzKYAAoAEpZmsoEQRQMFWHlN4ZAZQg2zdez2AAXlASzoDc2zgucnygsmVgQi9GDWXgwNEPyMKBWWOVZHivNEuSn1YP9BCCKyqU4x8xoACEtAsC5oGmPKcwSpLoOK1KAyDUAwqI1NCD4aqyMoH9izuSjqNojAcHMpi-FaJg+NYd1LzqEwNMappW1YSBuAQEIgA)

**Review by jakebailey — CHANGES_REQUESTED** [review]

**Review by jakebailey — APPROVED** [review]

**Comment by iamwinner422** [comment]
Ready to merge

**Pull request merged**